modules/guardian/msgs.go

```diff
@@ -33,30 +32,142 @@ func (msg MsgAddProfiler) GetSignBytes() []byte {
+func (msg MsgDeleteProfiler) GetSignBytes() []byte {
+	b, err := msgCdc.MarshalJSON(msg)
+	if err != nil {
+		panic(err)
 	}
-	if len(msg.AddedAddr) == 0 {
-		return sdk.ErrInvalidAddress(msg.AddedAddr.String())
+	return sdk.MustSortJSON(b)
+}
+
+func (msg MsgDeleteProfiler) ValidateBasic() sdk.Error {
+	return msg.DeleteGuardian.ValidateBasic()
+}
+
+func (msg MsgDeleteProfiler) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.DeletedAddress}
+}
+
+//______________________________________________________________________
+// MsgAddTrustee - struct for add a trustee
+type MsgAddTrustee struct {
+	AddGuardian
+}
+
+func NewMsgAddTrustee(description string, accountType AccountType, address, addedAddress sdk.AccAddress) MsgAddTrustee {
+	return MsgAddTrustee{
+		AddGuardian: AddGuardian{
+			Description:  description,
+			Address:      address,
+			AddedAddress: addedAddress,
+		},
 	}
-	return nil
+}
+func (msg MsgAddTrustee) Route() string { return MsgType }
+func (msg MsgAddTrustee) Type() string  { return "guardian add-trustee" }
+func (msg MsgAddTrustee) GetSignBytes() []byte {
+	b, err := msgCdc.MarshalJSON(msg)
+	if err != nil {
+		panic(err)
+	}
+	return sdk.MustSortJSON(b)
 }
 
-func (msg MsgAddProfiler) GetSigners() []sdk.AccAddress {
-	return []sdk.AccAddress{msg.AddedAddr}
+func (msg MsgAddTrustee) ValidateBasic() sdk.Error {
+	return msg.AddGuardian.ValidateBasic()
+}
+
+func (msg MsgAddTrustee) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.AddedAddress}
 }
 
 //______________________________________________________________________
+// MsgDeleteTrustee - struct for delete a trustee
+type MsgDeleteTrustee struct {
+	DeleteGuardian
+}
 
-func validName(name string) bool {
-	if len(name) == 0 || len(name) > 128 {
-		return false
+func NewMsgDeleteTrustee(address, deletedAddress sdk.AccAddress) MsgDeleteTrustee {
+	return MsgDeleteTrustee{
+		DeleteGuardian: DeleteGuardian{
+			Address:        address,
+			DeletedAddress: deletedAddress,
+		},
+	}
+}
+func (msg MsgDeleteTrustee) Route() string { return MsgType }
+func (msg MsgDeleteTrustee) Type() string  { return "guardian delete-trustee" }
+func (msg MsgDeleteTrustee) GetSignBytes() []byte {
+	b, err := msgCdc.MarshalJSON(msg)
+	if err != nil {
+		panic(err)
 	}
+	return sdk.MustSortJSON(b)
+}
+
+func (msg MsgDeleteTrustee) ValidateBasic() sdk.Error {
+	return msg.DeleteGuardian.ValidateBasic()
+}
+
+func (msg MsgDeleteTrustee) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.DeletedAddress}
+}
 
-	// Must contain alphanumeric characters, _ and - only
-	reg := regexp.MustCompile(`[^a-zA-Z0-9_-]`)
-	return !reg.Match([]byte(name))
+//______________________________________________________________________
+
+type AddGuardian struct {
+	Description  string         `json:"description"`
+	Address      sdk.AccAddress `json:"address"`
+	AddedAddress sdk.AccAddress `json:"added_address"`
+}
```

Comment:
Description  string
Address       sdk.AccAddress     // address added
AddedBy     sdk.AccAddress     // address that initiated the tx